src/Groups/homomorphisms.jl

```diff
@@ -386,20 +386,21 @@ function hom(x::GAPGroupElem{AutomorphismGroup{T}}) where T <: GAPGroup
   return _hom_from_gap_map(G, G, x.X)
 end
 
-(f::GAPGroupElem{AutomorphismGroup{T}})(x::GAPGroupElem{T}) where T <: GAPGroup = apply_automorphism(f, x)
-Base.:^(x::GAPGroupElem{T},f::GAPGroupElem{AutomorphismGroup{T}}) where T <: GAPGroup = apply_automorphism(f, x)
+(f::GAPGroupElem{AutomorphismGroup{T}})(x::GAPGroupElem) where T <: GAPGroup = apply_automorphism(f, x, true)
+Base.:^(x::GAPGroupElem{T},f::GAPGroupElem{AutomorphismGroup{T}}) where T <: GAPGroup = apply_automorphism(f, x, true)
+#Base.:^(f::GAPGroupElem{AutomorphismGroup{T}},g::GAPGroupElem{AutomorphismGroup{T}}) where T <: GAPGroup = g^-1*f*g
 
 function (A::AutomorphismGroup{T})(f::GAPGroupHomomorphism{T,T}) where T <: GAPGroup
    @assert domain(f)==A.G && codomain(f)==A.G "f not in A"
    @assert isbijective(f) "f not in A"
    return group_element(A, f.map)
 end
 
-function apply_automorphism(f::GAPGroupElem{AutomorphismGroup{T}}, x::GAPGroupElem{T}, check=true) where T <: GAPGroup
+function apply_automorphism(f::GAPGroupElem{AutomorphismGroup{T}}, x::GAPGroupElem, check=true) where T <: GAPGroup
   A = parent(f)
   G = parent(x)
   if check
-    @assert A.G == G || GAP.Globals.IN(x.X, A.G.X) "Not in the domain of f!"
+    @assert A.G == G || GAP.Globals.IN(x.X, A.G.X) "Not in the domain of f!"      #TODO Do we really need the IN check?
```

Comment:
Concerning the question:
Without the `IN` check, GAP will show an error message that is not helpful.
We can argue that GAP should give a better error message, and then the `IN` test can be omitted.

Reply:
The idea is to allow to compute homomorphisms only on elements of the domain, even if it makes sense for larger sets (e.g. conjugator automorphisms defined only on proper subgroups). The `check` variable was not intended to be used by the upper level user, instead by the programmer who writes an algorithm involving automorphisms. If he wants to apply automorphisms on elements that are sure to belong to the domain, he can deactivate the `check` in order to avoid the `IN` check, that could be a quite expensive procedure.